packages/common/src/ContractInteractor.ts

```diff
@@ -565,9 +567,17 @@ export class ContractInteractor {
   async getBlockNumber (): Promise<number> {
     let blockNumber = -1
     let attempts = 0
-    while (blockNumber < this.lastBlockNumber && attempts <= 100) {
-      blockNumber = await this.web3.eth.getBlockNumber()
-      await sleep(100)
+    const delay = this.chain === 'private' ? 0 : 1000
```

Comment:
It also seems you never actually care about 'chain' aka 'network type' but use it to decide what delay to use. Maybe you can just have a `this.rpcRetryDelay` member or something like that? Would make it all more obvious.